### ISSUE TITLE
extract plugin improvement suggestion

### ISSUE THREAD

**Issue by bchevalier** [pixijs/pixijs#8114]
Hello,

This is to suggest an improvement to the extract plugin API.

What I am trying to achieve is fetching a single pixel from an offscreen rendered container. For this purpose, I use the extract image API.

By extracting the image and then rendering it into a canvas I can then read the desired pixel. However, I only need a single pixel from the image and therefore this solution is overkill.

Here is the current usage of the `extract.image` API:
```js
const image = renderer.plugins['extract'].image(container);
```

The `extract.pixels` API is probably faster but still overkill. However it is made unusable by the fact that it is impossible to know the dimensions of the image that the pixels represent:
```js
const pixels = renderer.plugins['extract'].pixels(container);
```
The `pixels` object is an array, it contains no information on width or height.

Ideally, the `extract.pixels` and `extract.image` APIs would accept a scale, a position and a dimension:
```js
const pixels = renderer.plugins['extract'].pixels(spineView.parent, scale, x, y, width, height);
```
The position and dimension would determine which part of the container to clip for extraction purposes. And the scale would define the scaling ratio of the rendering. For instance a 200x300 clipped extraction with a scale of 0.5 would generate a pixel buffer corresponding to a 100x150px dimension.

**Comment by GoodBoyDigital** [comment]
this would be cool. Thinking about it, it feels like we could make use of our existing pixi object and terms:

`const pixels = renderer.plugins['extract'].pixels(spineView.parent, scale, x, y, width, height);`

becomes:

 ```
const bounds = new Rectangle(0,0,10,10);
const resolution = 0.5;

const pixels = renderer.plugins['extract'].pixels(spineView.parent, resolution, bounds);
```

the bounds and the resolution would be useful options for pretty much any extract function!
We use the `renderer.generateTexture` underneath which accepts both those properties so should be fairly easy to implement!

**Comment by pjalsGit** [comment]
Shouldn't it be Good First Issue instead of Good First PR?

**Comment by nouhmaster** [comment]
Hi we are a students in computer science in a Paris University,
We would try to correct this issues, if you are OK.

**Comment by bigtimebuddy** [comment]
Thank you @nouhmaster, PR welcome!

**Comment by nouhmaster** [comment]
hi 
we know you want this `const pixels = renderer.plugins['extract'].pixels(spineView.parent, scale, x, y, width, height);`
but in typeScript is not possible to have paramettre optionnal in first 
So the function if we change will look like  `const pixels = renderer.plugins['extract'].pixels(scale, x, y, width, height,spineView.parent);`
Did you have other option?

**Comment by bchevalier** [comment]
exactly, as suggested above: https://github.com/pixijs/pixijs/issues/8114#issuecomment-1021993657

**Comment by nouhmaster** [comment]
Ok Ok , thank you for your reply , why go to try this 😃

**Comment by nouhmaster** [comment]
hi 
I cannot fork the project because i am not a member , it is possible to have authorization please ? 🙂 .

**Comment by darnuria** [comment]
Hi, this issue was addressed by https://github.com/pixijs/pixijs/pull/8247 and get merged still open? :)
